test/org/centos/contra/Infra/Utils.groovy

```diff
@@ -0,0 +1,347 @@
+import spock.lang.*
+import com.homeaway.devtools.jenkins.testing.JenkinsPipelineSpecification
+
+import org.yaml.snakeyaml.Yaml
+
+public class UtilsSpec extends JenkinsPipelineSpecification {
+    @Shared
+    def config = new HashMap<String, ?>()
+
+    @Shared
+    def infraUtils
+
+    def setup() {
+        infraUtils = loadPipelineScriptForTest("../../src/org/centos/contra/Infra/Utils.groovy")
+        explicitlyMockPipelineStep('dir')
+        explicitlyMockPipelineStep('writeFile')
+        explicitlyMockPipelineStep('fileExists')
+        explicitlyMockPipelineStep('withEnv')
+        explicitlyMockPipelineStep('container')
+
+        explicitlyMockPipelineVariable('WORKSPACE')
+        explicitlyMockPipelineVariable('CREDENTIAL_FILE_NAME')
+        explicitlyMockPipelineVariable('SSH_FILE_NAME')
+    }
+
+    def setupSpec() {
+        Yaml yaml = new Yaml()
+        config = (Map<String, ?>) yaml.load(new FileReader('test/contra-sample.yml'))
+    }
+
+    def "create aws instances"() {
+        def aws = config.infra.provision.cloud.aws
```

Comment:
All code should be within a block. This should go in a setup block.

Reply:
The `setup` block is allowed to be implicit, but I don't mind making them all explicit.

Reply:
I think it makes it easier to read and understand. I actually didn't realize it could be implicit, so thanks for explaining that!